tensorflow/core/user_ops/chol_grad.cc

```diff
@@ -0,0 +1,115 @@
+
+#include "tensorflow/core/framework/op_kernel.h"
+
+#include "third_party/eigen3/unsupported/Eigen/CXX11/Tensor"
+#include "tensorflow/core/framework/tensor_types.h"
+#include "tensorflow/core/framework/types.h"
+
+using namespace tensorflow;
+
+template <typename T>
+class CholeskyGrad : public OpKernel {
+    public:
+    
+    explicit CholeskyGrad(OpKernelConstruction* context) : OpKernel(context) {}
+
+    using Matrix = Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>;
+    using ConstMatrixMap = Eigen::Map<const Matrix>;
+    using MatrixMap = Eigen::Map<Matrix>;
+    using ConstRef = Eigen::Ref<const Matrix>;
+    using Ref = Eigen::Ref<Matrix>;
+    using lcl_size_t = int;
+
+    void Compute(OpKernelContext* context) override {
+
+    const Tensor& input_tensor_l = context->input(0); ;
+    const Tensor& input_tensor_l_bar = context->input(1); 
+
+    //Check that input tensors represent a matrix.
+    OP_REQUIRES(context, TensorShapeUtils::IsMatrix(input_tensor_l.shape()), errors::InvalidArgument("In[0] is not a matrix"));
+    OP_REQUIRES(context, TensorShapeUtils::IsMatrix(input_tensor_l_bar.shape()), errors::InvalidArgument("In[1] is not a matrix"));
+
+    //Check that input tensors are square.
+    OP_REQUIRES(context, input_tensor_l.dim_size(0) == input_tensor_l.dim_size(1), errors::InvalidArgument("Input matrix must be square."));
+    OP_REQUIRES(context, input_tensor_l_bar.dim_size(0) == input_tensor_l_bar.dim_size(1), errors::InvalidArgument("Input matrix must be square."));
+
+    //Check that input tensors are of same size.
+    OP_REQUIRES(context, input_tensor_l.dim_size(0) == input_tensor_l_bar.dim_size(0), errors::InvalidArgument("Input matrices must be of same size."));    
+
+    // Create an output tensor
+    Tensor* output_tensor = NULL;
+    
+    OP_REQUIRES_OK(context, context->allocate_output(0, input_tensor_l_bar.shape(), &output_tensor));
+
+    if (output_tensor->NumElements() == 0) {
+      // the output shape is a 0-element matrix, so there is nothing to do.
+      return;
+    }
+    
+    //The next three lines are necessary to get Eigen matrix behaviour.
+    const ConstMatrixMap input_matrix_l(input_tensor_l.flat<T>().data(), input_tensor_l.dim_size(0), input_tensor_l.dim_size(1));
+    const ConstMatrixMap input_matrix_l_bar(input_tensor_l_bar.flat<T>().data(), input_tensor_l_bar.dim_size(0), input_tensor_l_bar.dim_size(1));
+    MatrixMap output_matrix(output_tensor->template flat<T>().data(), input_tensor_l.dim_size(0), input_tensor_l.dim_size(1) );    
+
+    const lcl_size_t N = input_matrix_l.rows();
+    const lcl_size_t NB = 32;
+
+    output_matrix = input_matrix_l_bar.template triangularView<Eigen::Lower>();
+
+    
+    for ( lcl_size_t Ji = (N-NB+1) ; Ji>(1-NB); Ji-= NB )   
```

Comment:
A few general code style guidelines:
  * no space inside parens. 
  * space after comma and on both sides of binary operators. 
  * curly braces at the end of line,
  * use descriptive variable names.

You can see the full set of guidelines here: http://google.github.io/styleguide/cppguide.html

I would change the variable names to:

NB -> kMaxBlockSize
Ji ->  block_end (and redefined to point at the first element to the right of the current panel as is common with iterators in C++)
J -> block_begin
JB -> block_size

I guess this was "translated" from one-based to zero-based indexing and the subtraction of 1 everywhere makes the code harder to read. To improve readability I would suggest using something like the following:

for (int64 block_end = n; block_end > 0; block_end -= kMaxBlockSize) {
  const int64 block_begin = std::max(0ll, block_end - kMaxBlockSize);
  const int64 block_size = block_end - block_begin;
  const int64 trailing_size = n - block_end;
  output_matrix.block(block_end, block_begin, trailing_size, block_size) = ...
  output_matrix.block(block_begin, block_begin, block_size, block_size) = ...
  etc.